editor/components/post-publish-panel/index.js

```diff
@@ -41,6 +46,12 @@ class PostPublishPanel extends Component {
 				loading: false,
 			} );
 		}
+
+		// Automatically collapse the publish sidebar when
+		// a post is published and the user makes an edit.
+		if ( this.props.isPublished && newProps.isDirty ) {
```

Comment:
Minor: [`componentWillReceiveProps` is slated for deprecation](https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md) in favor of either `static getDerivedStateFromProps` or `componentDidUpdate`. Since this doesn't classify as a state derivation, it'd be most future-proof to include this logic in a `componentDidUpdate`.